Lesson5

amzhulanov opened the pull request:
ДЗ выполнено.
Надеюсь я правильно понял, что "сквозная индексация массивов" должна быть при расчёте математической формулы.

amzhulanov committed:
ДЗ выполнено. Создал три класса.
CalcSpeed - класс с методом main().
В нём вызывается метод заполнения массива.
Обработка массива одним потоком
Обработка массива двумя потоками

amzhulanov committed:
Скорректировал вид
Ctrl+Alt+L

amzhulanov committed:
Добавил заполнение исходного массива дефолтными значениями, после обработки однопотоковым методом.
Убрал сквозную адресацию, а для корректности расчёта математической формулы сделал (n+h)

amzhulanov committed:
ДЗ выполнено. Создал три класса.
CalcSpeed - класс с методом main().
В нём вызывается метод заполнения массива.
Обработка массива одним потоком
Обработка массива двумя потоками

amzhulanov committed:
Незначительные правки

usharik commented on src/Java/ru/geekbrains/lesson5/MultiThreading.java:
```diff
@@ -0,0 +1,54 @@
+
+        firstThread();
+        secondThread();
+        while (!waitFirstThread || !waitSecondThread) {
```
Так можно, но правильней было бы использовать join.

amzhulanov replied:
Как раз сейчас переделываю. Хочу добавить join() и Runnable

usharik commented on src/Java/ru/geekbrains/lesson5/MultiThreading.java:
```diff
@@ -0,0 +1,54 @@
+        }
+        System.arraycopy(arrMultiThread1, 0, arrOneThread, 0, h);
+        System.arraycopy(arrMultiThread2, 0, arrOneThread, h, h);
+        delay = System.currentTimeMillis() - a;
```
Переменную delay можно было бы вообще не объявлять, а сразу вернуть разницу

usharik commented on src/Java/ru/geekbrains/lesson5/MultiThreading.java:
```diff
@@ -0,0 +1,54 @@
+            public void run() {
+                for (int i = 0;
+                     i < h - 1; i++) {
+                    arrMultiThread1[i] = (float) (arrMultiThread1[i] * Math.sin(0.2f + i / 5) * Math.cos(0.2f + i / 5) * Math.cos(0.4f + i / 2));
```
Вы создали два почти одинаковых класса. Можно обойтись одним, если передавать параметры (массив и смещение первого индекса) через конструктор.

amzhulanov committed:
1. вынес расчёт матиматическй формулы в отдельный метод
2. заменил цикл while при ожидании завершения потоков на join
3. стал использовать Runabble

amzhulanov committed:
Убрал дублирование.
Удалил один избыточный класс

amzhulanov committed:
Поправил метод заполнения массива. Добавил передачу параметров при вызове метода.
